src/vec.rs

```diff
@@ -859,25 +899,75 @@ impl<T, const N: usize> Vec<T, N> {
         }
 
         // Stage 1: Nothing was deleted.
-        process_loop::<F, T, N, false>(original_len, &mut f, &mut g);
+        process_loop::<F, T, N, A, false>(original_len, &mut f, &mut g);
 
         // Stage 2: Some elements were deleted.
-        process_loop::<F, T, N, true>(original_len, &mut f, &mut g);
+        process_loop::<F, T, N, A, true>(original_len, &mut f, &mut g);
 
         // All item are processed. This can be optimized to `set_len` by LLVM.
         drop(g);
     }
 }
 
+impl<const N: usize, A> Vec<u8, N, A> {
+    /// Transmutes the filled buffer to the output type. The storage of the length of the [`Vec`] does
+    /// not participate in the transmutation.
+    ///
+    /// # Safety
+    ///
+    /// - The buffer must be full.
+    /// - The size of the output type must be equal to the size of the **buffer** (rather than [`Vec`]).
+    /// - The alignment of the `Vec` must be equal or be a multiple of the alignment of the output.
+    ///
+    /// # Panics
+    ///
+    /// In debug mode, this method panics...
+    ///
+    /// - If the buffer isn't full.
+    /// - If the size of the output type doesn't match the size of the buffer.
+    /// - If the alignment of the `Vec` isn't equal or a multiple of the alignment of the output.
```

Comment:
I don't think these panics should be documented, given they only happen in debug mode.